src/main/java/com/stripe/net/LiveStripeResponseGetter.java

```diff
@@ -502,20 +510,20 @@ private static StripeResponse getMultipartStripeResponse(
 
 	}
 
-	private static void handleAPIError(String rBody, int rCode)
+	private static void handleAPIError(String rBody, int rCode, String requestId)
 			throws InvalidRequestException, AuthenticationException,
 			CardException, APIException {
 		LiveStripeResponseGetter.Error error = APIResource.GSON.fromJson(rBody,
 				LiveStripeResponseGetter.ErrorContainer.class).error;
 		switch (rCode) {
 		case 400:
-			throw new InvalidRequestException(error.message, error.param, null);
+			throw new InvalidRequestException(error.message, requestId, error.param, null);
 		case 404:
-			throw new InvalidRequestException(error.message, error.param, null);
+			throw new InvalidRequestException(error.message, requestId, error.param, null);
 		case 401:
 			throw new AuthenticationException(error.message);
 		case 402:
-			throw new CardException(error.message, error.code, error.param, error.decline_code, error.charge, null);
+			throw new CardException(error.message, requestId, error.code, error.param, error.decline_code, error.charge, null);
 		default:
 			throw new APIException(error.message, null);
```

Comment:
Do we never return a request ID on 500s?  We definitely record the iar in some cases.